src/rustdoc_gen.rs

```diff
@@ -455,13 +468,32 @@ impl RustdocGenerator for RustdocFromProjectRoot {
                 }
             }
         }).into_terminal_result()?;
+
+        Ok(Some(generate_data_request(
+            config,
+            CrateSource::ManifestPath { manifest },
+            crate_data,
+        )))
```

Comment:
I notice this part is the same across all three implementations that actually generate data.

It's always:
```rust
        Ok(Some(generate_data_request(
            config,
            // *some* `CrateSource` variant
            crate_data,
        )))
```

Can we keep refactoring this stuff down to where we just return `CrateSource` here, and then plug in the `config` and `crate_data` somewhere else in a common layer across all three implementations? That layer can then do the `Ok(Some(generate_data_request(..)))` wrapping too, if needed.